### PR TITLE
Complete website with navigation (Single Page Application)

### PR DESCRIPTION
**Pull request by Jord4N-AC**
This project was made pair-programming, In this project, I and my coding partners have converted the Awesome books application into a fully functional website. We refactored the project to add a new Contact info section, a Navigation bar, and a Footer.

### Project requirements and Features:

- Add a Navigation bar with 3 links (Books list, Add book form, Contact info)
- When clicking on any of the links, the main area changes to display only one of the three content sections
- Use only JavaScript and CSS, no libraries or frameworks

### General Requirements:

- Check linter errors
- Use the correct GitHub flow
- Document my work in a professional way
- Follow best practices for HTML, CSS & JavaScript
- Open a pull request and ask for a review

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,32 +1,60 @@
...
+  <body>
+    <nav class="navbar">
+      <div>
+        <h3 id="bar-header">Awesome Books</h3>                
+      </div>
+      <div class="nav-list">
+        <a id="list-link" href="#books-header">List</a>
+        <a id="add-new-link" class="add-new" href="#form">Add New</a>
+        <a id="contact-link" href="#contact">Contact</a>
+      </div>
```
Comment (CollinsTatang) [inline index.html:20]: - [ ] Please when any of the links are clicked, the main area changes to display only one of the three content sections. 
### Please check the project [requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/books/m4_full_website.md#project-requirements) again as you make the correction

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi @Jord4N-AC, @AbrahaKahsay and @Brenda309 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ] Please all partners should make multiple commits to the projects.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: index.html (hunk 1 above)
Comment (Jord4N-AC) [reply index.html:20]: @CollinsTatang I don't know what you really mean by 'when any of the links are clicked, the main area changes to display only one of the three content sections'
[here](https://jord4n-ac.github.io/Awesome-books/#books-header) is the live link of this project, it is working as the requirements ask
I don't really understand your feedback or what you want me to change

Comment (CollinsTatang) [reply index.html:20]: ### You can check the wire-frame and see how the different nav link display  
frame

Comment (Jord4N-AC) [reply index.html:20]: @CollinsTatang
I did check the requirements and make sure to meet them.
I'm not sure, but I guess is a deployment problem, When I try it on my browser everything works well, but when I share the link to someone else It doesn't work, I don't really know how to solve this

Comment (CollinsTatang) [reply index.html:20]: That means you deploy before making recent changes 
Try to deploy the project on notify

Comment (CollinsTatang) [reply index.html:20]: Push all your changes before deployment

Comment (CollinsTatang) [reply index.html:20]: - [ ] Please contact your partners to work with you and each and everyone should make multiple commits

Comment (Jord4N-AC) [reply index.html:20]: I deployed after the changes when I finished the project, So I don't really know why this is not working.

Comment (Brenda309) [reply index.html:20]: Thank you, I think that will help.

Comment (CollinsTatang) [reply index.html:20]: Check the branch you deployed

Comment (Jord4N-AC) [reply index.html:20]: I deployed on the featured branch
I have tried everything but this still doesn't work

Comment (Jord4N-AC) [reply index.html:20]: > Check the branch you deployed
I deployed on the featured branch

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @Jord4N-AC and Team 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
#### I can not highlight the code because this PR does not include that code as changed

- [ ] The method in the `index.js` file line 82 is trying to get data from the local storage but there is an issue, If a user runs your project and this function will be called but on the user's browser there's no data with `'booksArray'` object and it will return null so the line 83 will throw an error, no single functionality will work because that forEach loop can not iterate a null value. You must check if there's data on the browser before returning that data. Please fix this issue.
 - To fix this you can consider to replace the method with this code.
 `static loadBooks() {
    const data =JSON.parse(localStorage.getItem('booksArray'));
    if(data != null) {
    data.forEach((book, i) => {
      const newBook = new BookObject(book.title, book.author, i);
      newBook.add();
      theBooks.push(newBook);
    });
    }
  }`
  
### See the image of the error.

image

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AbrahaKahsay** [comment]
@thecodechaser Thank you so much. your suggestion was very helpful.

**Comment by Brenda309** [comment]
Thank you so @thecodechaser, for such a helpful review.
For our next reviewer, everything is working now as expected. Just a small clarification on the book "list", the books appear when you first added them. That means if you want to see the list of books please add them first.

**Review by brytebee — APPROVED** [review]
Hi, @Team👋,

# STATUS: APPROVED 🕺🥳 

Congratulations🍾🥳!! Your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 



Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**